extensions/oidc/runtime/src/main/java/io/quarkus/oidc/runtime/OidcTenantConfig.java

```diff
@@ -1188,6 +1188,13 @@ interface Token {
          */
         Optional<String> header();
 
+        /**
+         * If this OIDC tenant should be resolved by presence of the custom HTTP header that contains a bearer token.
+         * This option is valid only when the {@link Token#header()} configuration property is set.
+         */
+        @WithDefault("false")
+        boolean resolveTenantWithHeader();
```

Comment:
I wonder, should we do it by default... Currently, the only default option, before we fallback to the default tenant,  is to check if the request path has a `tenant-id` property in it.

If a user configured `quarkus.oidc.token.header=My-Token` and no other tenant resolvers can resolve it, then as a last try, we can try the header match, before choosing a default tenant ?

Reply:
Done.

Reply:
argh, sorry for couple of commits, but I misunderstood what priority you meant for this new resolver. I think now it is fine, let's see.